internal/cephfs/controllerserver.go

```diff
@@ -173,6 +211,17 @@ func checkValidCreateVolumeRequest(
 				parentVol.Size,
 				vol.Size)
 		}
+
+		if vol.BackingSnapshot {
+			volCaps := req.GetVolumeCapabilities()
+			for _, volCap := range volCaps {
```

Comment:
Also, need to block restore to the bigger size PVC?

Reply:
@Madhu-1 You're right, the size should equal to that of the parent snapshot (or subvolume, as it is in this case). Although I think there is a bug here, even in the original code: we're taking the size of the whole subvolume, not the size of the snapshot. But the size of the original PVC might have changed since the snapshot was taken.

Reply:
Yes looks like it's a bug. let me experiment on this one and fix it.

Reply:
https://github.com/ceph/ceph-csi/issues/3147 here is the one.

Reply:
Cool, thanks!